pkgs/by-name/mc/mcomix/package.nix

```diff
@@ -8,37 +8,53 @@
 , testers
 , wrapGAppsHook
 
-# Recommended Dependencies:
-, lhasa
-, mupdf
+  # Recommended Dependencies:
 , p7zip
 , unrar
+, chardetSupport ? true
+, pdfSupport ? true
 , unrarSupport ? false  # unfree software
 }:
 
 python3.pkgs.buildPythonApplication rec {
   pname = "mcomix";
-  version = "2.2.1";
+  version = "3.0.0";
+  pyproject = true;
 
   src = fetchurl {
     url = "mirror://sourceforge/mcomix/${pname}-${version}.tar.gz";
-    hash = "sha256-fmnlPhNCN6YR3lW2YCMEAbEiWVigcfFDq1tDQ1eTNkA=";
+    hash = "sha256-InDEPXXih49k5MiG1bATElxCiUs2RZTV7JeRVMTeoAQ=";
   };
 
-  buildInputs = [ gtk3 gdk-pixbuf ];
-  nativeBuildInputs = [ wrapGAppsHook gobject-introspection ];
-  propagatedBuildInputs = (with python3.pkgs; [ pillow pygobject3 pycairo ]);
+  buildInputs = [
+    gtk3
+    gdk-pixbuf
+  ];
 
-  # Tests are broken
+  nativeBuildInputs = [
+    gobject-introspection
+    python3.pkgs.setuptools
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pillow
+    pycairo
+    pygobject3
+  ]
+  ++ lib.optionals chardetSupport [ chardet ]
+  ++ lib.optionals pdfSupport [ pymupdf ];
+
+  # No tests included in .tar.gz
   doCheck = false;
 
-  # prevent double wrapping
+  # Prevent double wrapping
   dontWrapGApps = true;
 
   preFixup = ''
     makeWrapperArgs+=(
       "''${gappsWrapperArgs[@]}"
-      "--prefix" "PATH" ":" "${lib.makeBinPath ([ p7zip lhasa mupdf ] ++ lib.optional (unrarSupport) unrar)}"
+      "--prefix" "PATH" ":" "${lib.makeBinPath ([ p7zip ] ++ lib.optional unrarSupport unrar)}"
```

Comment:
was those removed?

Reply:
- `mupdf` was replaced with `pymupdf`
- if `p7zip` is installed it is preferred over `lhasa`: https://sourceforge.net/p/mcomix/git/ci/master/tree/mcomix/archive_tools.py. So doesn't make sense to include both